vendor/bat-native-ledger/src/bat/ledger/internal/publisher/publisher_server_list.cc

```diff
@@ -73,13 +84,20 @@ void PublisherServerList::OnDownload(
     int response_status_code,
     const std::string& response,
     const std::map<std::string, std::string>& headers,
-    DownloadServerPublisherListCallback callback) {
+    ledger::ResultCallback callback) {
   ledger_->LogResponse(
       __func__,
       response_status_code,
       "Publisher list",
       headers);
 
+  // we iterated through all pages
+  if (response_status_code == net::HTTP_NO_CONTENT) {
```

Comment:
Do we always have to make an extra network request to know that we're done? Typically a paging JSON endpoint would return a "more_remaining" field or something like that.

Reply:
we wanted to do as minimal change as possible, so that we would not need to change server side a lot. I agree that not doing an extra request would be better, but with v4 in sight I would not put more work on server side